src/ImageRunModal.jsx

```diff
@@ -27,13 +28,107 @@ const units = {
+                                <div role='group' className='ct-form-layout-split'>
+                                    <input className='form-control'
+                                           id={ (idx === 0 && id) || '' }
+                                           type='number'
+                                           step={1}
+                                           min={1}
+                                           max={65535}
+                                           placeholder={_("Container port")}
+                                           value={port.containerPort || ''}
+                                           onChange={e => this.onPortChange(idx, 'containerPort', e.target.value)} />
+                                    <Select.Select className='form-control'
+                                                   initial={port.protocol}
+                                                   onChange={value => this.onPortChange(idx, 'protocol', value)} >
+                                        <Select.SelectEntry data='TCP' key='TCP'>
+                                            {_("TCP")}
+                                        </Select.SelectEntry>
+                                        <Select.SelectEntry data='UDP' key='UDP'>
+                                            {_("UDP")}
+                                        </Select.SelectEntry>
+                                    </Select.Select>
+                                    <input className='form-control'
+                                           type='number'
+                                           step={1}
+                                           min={1}
+                                           max={65535}
+                                           placeholder={_("Host port")}
+                                           value={port.hostPort || ''}
+                                           onChange={e => this.onPortChange(idx, 'hostPort', e.target.value)} />
+                                </div>
+                                <div role='group' className='ct-form-layout-split'>
+                                    <Button bsStyle='default' className='pficon-close'
+                                            disabled={dialogValues.publish.length === 1 &&
+                                                      (!dialogValues.publish[0]['containerPort'] ||
+                                                       !dialogValues.publish[0]['hostPort'])}
+                                            onClick={() => this.removePort(idx)} />
+                                    <Button bsStyle='default' className='fa fa-plus' onClick={this.addPort} />
+                                </div>
+                            </div>
+                        )
+                    )
+                }
+            </React.Fragment>
+        );
+    }
+}
+PublishPortsForm.propTypes = {
```

Comment:
You 're missing here id: PropTypes.string.isRequired

But no need to repost for this I guess.

Reply:
The id is only required on the first row/instance.

Reply:
Right, but it should still be always given with a PublishPortsForm instantiation, no?

Reply:
oh woops, you're right. sorry about that :/

Reply:
Fixed it!